xdsl/dialects/builtin.py

```diff
@@ -232,27 +238,20 @@ def from_params(
 AnyIntegerAttr: TypeAlias = IntegerAttr[IntegerType | IndexType]
 
 
+@irdl_attr_definition
 class Float16Type(ParametrizedAttribute, MLIRType):
     name = "f16"
 
 
-f16 = Float16Type()
-
-
 @irdl_attr_definition
 class Float32Type(ParametrizedAttribute, MLIRType):
     name = "f32"
 
 
-f32 = Float32Type()
```

Comment:
This is removing functionality, i.e. accessing builtin.f32 to get an f32. We can consider removing this, though builtin.f32 might be more natural than builtin.Float32Type.

Reply:
Not sure I understand the functionality that is removed?
If so, we need a test/MFE to prove it?

Reply:
You won't be able to access the Float32Type anymore by just calling f32. Where we actually want this/need this functionality is a different question. Maybe @math-fehr  can decide. Me personally, I don't think we need this functionality, so I am fine with these changes.

Reply:
Lets wait for @math-fehr to see what to do

Reply:
I personally kind of like having those aliases somewhere.
This is just to write `f32` rather than `Float32Type()`, which I think makes it nicer to read IMO.
Maybe we should put them next to each other at the end of the file?

Reply:
ok, fixed. Maybe in some future PR we should also team-up somewhere stuff like:
```
AnyIntegerAttr, AnyFloat etc
```

Reply:
Yes, we should look into it!
Though we should be careful, since some of these constructs are also used in the file directly.